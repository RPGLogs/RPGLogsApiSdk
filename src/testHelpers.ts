import { buildSdk } from './main';
import btoa from 'btoa';

export function buildTestSdk(): ReturnType<typeof buildSdk> {
  return buildSdk(<string>process.env.TEST_ACCESS_TOKEN);
}

export async function getAccessToken(
  clientId: string,
  clientSecret: string,
  game = 'warcraft'
): Promise<string> {
  if (game === 'warcraft') game = 'www.warcraft';

  const authHeader = 'Basic ' + btoa(clientId + ':' + clientSecret);

  const response = await fetch(`https://${game}logs.com/oauth/token`, {
    method: 'POST',
    headers: {
      Authorization: authHeader,
      'Content-Type': 'application/x-www-form-urlencoded',
    },
    body: 'grant_type=client_credentials',
  });

  const json = await response.json();

  if (response.status === 200) {
    return json.access_token;
  } else {
    throw new Error('Response was not OK: ' + JSON.stringify(json ?? {}));
  }
}
