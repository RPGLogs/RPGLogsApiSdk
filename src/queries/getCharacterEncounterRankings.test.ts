import { buildTestSdk } from '../testHelpers';

test('returns data', async () => {
  const response = await buildTestSdk().getCharacterEncounterRankings({
    encounterId: 2333,
    characterId: 46159040,
  });

  expect(response?.characterData?.character).toMatchSnapshot();
});
