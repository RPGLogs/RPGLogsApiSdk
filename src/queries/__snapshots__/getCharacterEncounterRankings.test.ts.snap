// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns data 1`] = `
Object {
  "encounterRankings": Object {
    "averagePerformance": 68.0780088892181,
    "bestAmount": 255365,
    "difficulty": 5,
    "fastestKill": 79259,
    "medianPerformance": 71.49022998782577,
    "metric": "dps",
    "partition": 3,
    "ranks": Array [
      Object {
        "amount": 255365,
        "bestSpec": "Vengeance",
        "bracketData": 482,
        "duration": 79259,
        "guild": Object {
          "faction": 1,
          "id": 369203,
          "name": "Reaction",
        },
        "historicalPercent": 100,
        "historicalTotalParses": 4961,
        "lockedIn": true,
        "rankPercent": 100,
        "rankTotalParses": 4961,
        "report": Object {
          "code": "ANw6Fynq8W3P1Bkg",
          "fightID": 15,
          "startTime": 1602092886405,
        },
        "spec": "Vengeance",
        "startTime": 1602097935818,
        "todayPercent": 100,
        "todayTotalParses": 4977,
      },
      Object {
        "amount": 119065,
        "bestSpec": "Vengeance",
        "bracketData": 482,
        "duration": 159698,
        "guild": Object {
          "faction": 1,
          "id": 369203,
          "name": "Reaction",
        },
        "historicalPercent": 94.1402647302821,
        "historicalTotalParses": 4223,
        "lockedIn": true,
        "rankPercent": 94.1402647302821,
        "rankTotalParses": 4223,
        "report": Object {
          "code": "fKr2nkaQZgh8ptwx",
          "fightID": 28,
          "startTime": 1597256183915,
        },
        "spec": "Vengeance",
        "startTime": 1597263650997,
        "todayPercent": 92.77916666666667,
        "todayTotalParses": 4977,
      },
      Object {
        "amount": 118174,
        "bestSpec": "Vengeance",
        "bracketData": 482,
        "duration": 154152,
        "guild": Object {
          "faction": 1,
          "id": 369203,
          "name": "Reaction",
        },
        "historicalPercent": 94.00671895625695,
        "historicalTotalParses": 3780,
        "lockedIn": true,
        "rankPercent": 94.00671895625695,
        "rankTotalParses": 3780,
        "report": Object {
          "code": "vRyg7N9K2bpZmn3J",
          "fightID": 28,
          "startTime": 1596045585493,
        },
        "spec": "Vengeance",
        "startTime": 1596050506784,
        "todayPercent": 92.52313218390805,
        "todayTotalParses": 4977,
      },
      Object {
        "amount": 102224,
        "bestSpec": "Vengeance",
        "bracketData": 482,
        "duration": 124528,
        "guild": Object {
          "faction": 1,
          "id": 369203,
          "name": "Reaction",
        },
        "historicalPercent": 86.53504593883822,
        "historicalTotalParses": 4629,
        "lockedIn": true,
        "rankPercent": 86.53504593883822,
        "rankTotalParses": 4629,
        "report": Object {
          "code": "M7CLFvqzZJ26g4fQ",
          "fightID": 2,
          "startTime": 1598896777240,
        },
        "spec": "Vengeance",
        "startTime": 1598897179064,
        "todayPercent": 85.79066954745073,
        "todayTotalParses": 4977,
      },
      Object {
        "amount": 99913.6,
        "bestSpec": "Vengeance",
        "bracketData": 482,
        "duration": 148862,
        "guild": Object {
          "faction": 1,
          "id": 369203,
          "name": "Reaction",
        },
        "historicalPercent": 84.56445256683966,
        "historicalTotalParses": 4898,
        "lockedIn": true,
        "rankPercent": 84.56445256683966,
        "rankTotalParses": 4898,
        "report": Object {
          "code": "cPtj6y8nmAgD4rak",
          "fightID": 31,
          "startTime": 1600886616343,
        },
        "spec": "Vengeance",
        "startTime": 1600891559817,
        "todayPercent": 84.4341960029121,
        "todayTotalParses": 4977,
      },
      Object {
        "amount": 99480.6,
        "bestSpec": "Vengeance",
        "bracketData": 482,
        "duration": 167505,
        "guild": Object {
          "faction": 1,
          "id": 369203,
          "name": "Reaction",
        },
        "historicalPercent": 85.5069965458024,
        "historicalTotalParses": 4145,
        "lockedIn": true,
        "rankPercent": 85.5069965458024,
        "rankTotalParses": 4145,
        "report": Object {
          "code": "2Adwp1G76yJRL3qz",
          "fightID": 2,
          "startTime": 1597082905477,
        },
        "spec": "Vengeance",
        "startTime": 1597082951795,
        "todayPercent": 84.179974636575,
        "todayTotalParses": 4977,
      },
      Object {
        "amount": 94270.7,
        "bestSpec": "Vengeance",
        "bracketData": 482,
        "duration": 152267,
        "guild": Object {
          "faction": 1,
          "id": 369203,
          "name": "Reaction",
        },
        "historicalPercent": 81.64145485765974,
        "historicalTotalParses": 4382,
        "lockedIn": true,
        "rankPercent": 81.64145485765974,
        "rankTotalParses": 4382,
        "report": Object {
          "code": "bpzPRfCXKcDyTNL1",
          "fightID": 27,
          "startTime": 1597860136054,
        },
        "spec": "Vengeance",
        "startTime": 1597867805962,
        "todayPercent": 81.12115732368896,
        "todayTotalParses": 4977,
      },
      Object {
        "amount": 88940.1,
        "bestSpec": "Vengeance",
        "bracketData": 482,
        "duration": 93608,
        "guild": Object {
          "faction": 1,
          "id": 369203,
          "name": "Reaction",
        },
        "historicalPercent": 77.27003597251668,
        "historicalTotalParses": 4760,
        "lockedIn": true,
        "rankPercent": 77.27003597251668,
        "rankTotalParses": 4760,
        "report": Object {
          "code": "h36FTtvMyapcY9kg",
          "fightID": 25,
          "startTime": 1599674841193,
        },
        "spec": "Vengeance",
        "startTime": 1599679057949,
        "todayPercent": 77.05284377745998,
        "todayTotalParses": 4977,
      },
      Object {
        "amount": 83378.5,
        "bestSpec": "Vengeance",
        "bracketData": 481,
        "duration": 168555,
        "guild": Object {
          "faction": 1,
          "id": 369203,
          "name": "Reaction",
        },
        "historicalPercent": 74.6859870506115,
        "historicalTotalParses": 2931,
        "lockedIn": true,
        "rankPercent": 74.6859870506115,
        "rankTotalParses": 2931,
        "report": Object {
          "code": "3Q8tqnMNdZKzvmYg",
          "fightID": 25,
          "startTime": 1594231012083,
        },
        "spec": "Vengeance",
        "startTime": 1594234367791,
        "todayPercent": 71.61656856379525,
        "todayTotalParses": 4977,
      },
      Object {
        "amount": 81635.1,
        "bestSpec": "Vengeance",
        "bracketData": 482,
        "duration": 142274,
        "guild": Object {
          "faction": 1,
          "id": 369203,
          "name": "Reaction",
        },
        "historicalPercent": 71.49022998782577,
        "historicalTotalParses": 3245,
        "lockedIn": true,
        "rankPercent": 71.49022998782577,
        "rankTotalParses": 3245,
        "report": Object {
          "code": "WXP6wYcJG2RmaFQD",
          "fightID": 31,
          "startTime": 1594836297665,
        },
        "spec": "Vengeance",
        "startTime": 1594840193597,
        "todayPercent": 69.72736154177565,
        "todayTotalParses": 4977,
      },
      Object {
        "amount": 72266.5,
        "bestSpec": "Vengeance",
        "bracketData": 482,
        "duration": 104770,
        "guild": Object {
          "faction": 1,
          "id": 369203,
          "name": "Reaction",
        },
        "historicalPercent": 58.22399673927708,
        "historicalTotalParses": 4884,
        "lockedIn": true,
        "rankPercent": 58.22399673927708,
        "rankTotalParses": 4884,
        "report": Object {
          "code": "NyXmf1FZYK9kaHtn",
          "fightID": 41,
          "startTime": 1600279573912,
        },
        "spec": "Vengeance",
        "startTime": 1600716254246,
        "todayPercent": 58.05562738844353,
        "todayTotalParses": 4977,
      },
      Object {
        "amount": 71664.7,
        "bestSpec": "Vengeance",
        "bracketData": 476,
        "duration": 172698,
        "guild": Object {
          "faction": 1,
          "id": 369203,
          "name": "Reaction",
        },
        "historicalPercent": 61.783651713401625,
        "historicalTotalParses": 1730,
        "lockedIn": true,
        "rankPercent": 61.783651713401625,
        "rankTotalParses": 1730,
        "report": Object {
          "code": "txfY4hApvkj3HCzD",
          "fightID": 1,
          "startTime": 1592244507200,
        },
        "spec": "Vengeance",
        "startTime": 1592244732006,
        "todayPercent": 57.19104674884349,
        "todayTotalParses": 4977,
      },
      Object {
        "amount": 69915.1,
        "bestSpec": "Vengeance",
        "bracketData": 480,
        "duration": 188375,
        "guild": Object {
          "faction": 1,
          "id": 369203,
          "name": "Reaction",
        },
        "historicalPercent": 57.44566270011859,
        "historicalTotalParses": 1860,
        "lockedIn": true,
        "rankPercent": 57.44566270011859,
        "rankTotalParses": 1860,
        "report": Object {
          "code": "LjmKgCa34GNbA19c",
          "fightID": 26,
          "startTime": 1592416914988,
        },
        "spec": "Vengeance",
        "startTime": 1592425663986,
        "todayPercent": 54.677470333017276,
        "todayTotalParses": 4977,
      },
      Object {
        "amount": 68095.1,
        "bestSpec": "Vengeance",
        "bracketData": 482,
        "duration": 154647,
        "guild": Object {
          "faction": 1,
          "id": 369203,
          "name": "Reaction",
        },
        "historicalPercent": 52.55988111195291,
        "historicalTotalParses": 3518,
        "lockedIn": true,
        "rankPercent": 52.55988111195291,
        "rankTotalParses": 3518,
        "report": Object {
          "code": "p6HVdwvMA2T7XqKJ",
          "fightID": 6,
          "startTime": 1595441042430,
        },
        "spec": "Vengeance",
        "startTime": 1595442790350,
        "todayPercent": 52.0627532109301,
        "todayTotalParses": 4977,
      },
      Object {
        "amount": 67423.1,
        "bestSpec": "Vengeance",
        "bracketData": 482,
        "duration": 108960,
        "guild": Object {
          "faction": 1,
          "id": 369203,
          "name": "Reaction",
        },
        "historicalPercent": 51.088228949449935,
        "historicalTotalParses": 4947,
        "lockedIn": true,
        "rankPercent": 51.088228949449935,
        "rankTotalParses": 4947,
        "report": Object {
          "code": "KMqV9ZApxaQR7P3y",
          "fightID": 7,
          "startTime": 1601924511596,
        },
        "spec": "Vengeance",
        "startTime": 1601925813188,
        "todayPercent": 51.09731919662099,
        "todayTotalParses": 4977,
      },
      Object {
        "amount": 66312.1,
        "bestSpec": "Vengeance",
        "bracketData": 482,
        "duration": 101795,
        "guild": Object {
          "faction": 1,
          "id": 369203,
          "name": "Reaction",
        },
        "historicalPercent": 49.499117817006514,
        "historicalTotalParses": 4674,
        "lockedIn": true,
        "rankPercent": 49.499117817006514,
        "rankTotalParses": 4674,
        "report": Object {
          "code": "V4YfPm9h7FZq8GKx",
          "fightID": 29,
          "startTime": 1599069929729,
        },
        "spec": "Vengeance",
        "startTime": 1599074554524,
        "todayPercent": 49.473979243996666,
        "todayTotalParses": 4977,
      },
      Object {
        "amount": 65746.1,
        "bestSpec": "Vengeance",
        "bracketData": 481,
        "duration": 181446,
        "guild": Object {
          "faction": 1,
          "id": 369203,
          "name": "Reaction",
        },
        "historicalPercent": 51.20363589503429,
        "historicalTotalParses": 2590,
        "lockedIn": true,
        "rankPercent": 51.20363589503429,
        "rankTotalParses": 2590,
        "report": Object {
          "code": "xCz8BRdZW9hVL2fY",
          "fightID": 30,
          "startTime": 1593624601476,
        },
        "spec": "Vengeance",
        "startTime": 1593635973737,
        "todayPercent": 48.6164684493599,
        "todayTotalParses": 4977,
      },
      Object {
        "amount": 60595.1,
        "bestSpec": "Vengeance",
        "bracketData": 481,
        "duration": 170086,
        "guild": Object {
          "faction": 1,
          "id": 369203,
          "name": "Reaction",
        },
        "historicalPercent": 41.39043499840465,
        "historicalTotalParses": 2247,
        "lockedIn": true,
        "rankPercent": 41.39043499840465,
        "rankTotalParses": 2247,
        "report": Object {
          "code": "qxWj3dpFYMPhJBHQ",
          "fightID": 31,
          "startTime": 1593021592628,
        },
        "spec": "Vengeance",
        "startTime": 1593030303191,
        "todayPercent": 40.812514203469426,
        "todayTotalParses": 4977,
      },
      Object {
        "amount": 48619.4,
        "bestSpec": "Vengeance",
        "bracketData": 474,
        "duration": 273679,
        "guild": Object {
          "faction": 1,
          "id": 369203,
          "name": "Reaction",
        },
        "historicalPercent": 20.446372363865287,
        "historicalTotalParses": 1087,
        "lockedIn": true,
        "rankPercent": 20.446372363865287,
        "rankTotalParses": 1087,
        "report": Object {
          "code": "AYxJpkCtLazwfgdQ",
          "fightID": 2,
          "startTime": 1591639509560,
        },
        "spec": "Vengeance",
        "startTime": 1591640197358,
        "todayPercent": 22.05101459508469,
        "todayTotalParses": 4977,
      },
    ],
    "totalKills": 19,
    "zone": 24,
  },
  "id": 46159040,
  "name": "Xyä",
}
`;
